### PR TITLE
httpcli: env vars for retry params

### PR DESCRIPTION
**Pull request by tsenart**

### CODE REVIEW

File: internal/httpcli/client.go [hunk 1]
```diff
@@ -80,7 +80,12 @@ var redisCache = rcache.NewWithTTL("http", 604800)
 // and middleware pre-set for communicating with external services.
 var ExternalClientFactory = NewExternalClientFactory()
 
-var externalTimeout, _ = time.ParseDuration(env.Get("SRC_HTTP_CLI_EXTERNAL_TIMEOUT", "5m", "Timeout for external HTTP requests"))
+var (
+	externalTimeout, _          = time.ParseDuration(env.Get("SRC_HTTP_CLI_EXTERNAL_TIMEOUT", "5m", "Timeout for external HTTP requests"))
```
Comment (eseliger) [inline internal/httpcli/client.go:84]: should this use a `mustParseDuration`? Otherwise it'd be silently `0ns` I think. Might cause undesired behavior?

**Review by eseliger — APPROVED** [review]

**Review by efritz — APPROVED** [review]

File: internal/httpcli/client.go (hunk 1 above)
Comment (efritz) [reply internal/httpcli/client.go:84]: I agree that's probably wise.

Comment (tsenart) [reply internal/httpcli/client.go:84]: I made it so TimeoutOpt doesn't set a timeout if it's <= 0.

Comment (tsenart) [reply internal/httpcli/client.go:84]: I made it so TimeoutOpt doesn't set a timeout if it's <= 0.

**Pull request merged**